Agents/BMSUpdateAgent/BMSUpdateAgent/src/main/java/uk/ac/cam/cares/jps/agent/bmsupdate/BMSUpdateAgent.java

```diff
@@ -17,21 +18,40 @@
 import org.json.JSONObject;
 import uk.ac.cam.cares.jps.base.discovery.AgentCaller;
 import uk.ac.cam.cares.jps.base.exception.JPSRuntimeException;
+import uk.ac.cam.cares.jps.base.query.RemoteRDBStoreClient;
 import uk.ac.cam.cares.jps.base.query.RemoteStoreClient;
+import uk.ac.cam.cares.jps.base.timeseries.TimeSeries;
+import uk.ac.cam.cares.jps.base.timeseries.TimeSeriesClient;
 
 import static org.eclipse.rdf4j.sparqlbuilder.rdf.Rdf.iri;
 
+import java.sql.Connection;
+import java.text.SimpleDateFormat;
+import java.time.OffsetDateTime;
+import java.util.Date;
+import java.util.TimeZone;
+
 public class BMSUpdateAgent {
 
     private static final Logger LOGGER = LogManager.getLogger(BMSUpdateAgent.class);
 
     private final String STR_OM = "http://www.ontology-of-units-of-measure.org/resource/om-2/";
     private final Prefix P_OM = SparqlBuilder.prefix("om", iri(STR_OM));
 
-    final String FAIL_TO_GET_TEMP = "Fail to get temperature";
-    final String FAIL_TO_UPDATE_TEMP = "Fail to update temperature in knowledge graph";
-    final String FAIL_TO_TOGGLE = "Fail to trigger ESPHomeAgent to toggle device status";
-    final String FAIL_TO_PULL_DATA = "Fail to trigger ESPHomeUpdateAgent to pull data";
+    private final String STR_ONTOBMS = "https://www.theworldavatar.com/kg/ontobms/";
+    private final Prefix P_ONTOBMS = SparqlBuilder.prefix("ontobms", iri(STR_ONTOBMS));
+
+    final String FAIL_TO_GET_TEMP = "Fail to get temperature!";
+    final String FAIL_TO_UPDATE_TEMP = "Fail to update temperature in knowledge graph!";
+    final String FAIL_TO_TOGGLE = "Fail to trigger ESPHomeAgent to toggle device status!";
+    final String FAIL_TO_PULL_DATA = "Fail to trigger ESPHomeUpdateAgent to pull data!";
+    final String FAIL_TO_GET_BACNETDEVICEID = "Fail to get Bacnet Device ID!";
+    final String FAIL_TO_GET_BACNETOBJECTID = "Fail to get Bacnet Object ID!";
+    final String FAIL_TO_GET_LATESTDATA = "Unable to get latest timeseries data for the following IRI: ";
+    final String FAIL_TO_UPDATE_NEW_VALUE = "Unable to update new numerical value for the following IRI: ";
+
+    TimeSeries<OffsetDateTime> timeseries;
+    TimeSeriesClient<OffsetDateTime> tsClient;
```

Comment:
Should convert to local variables, since they are not shared across the class and only used in check latest data.